ncm-filecopy/src/main/pan/components/filecopy/schema.pan

```diff
@@ -22,21 +22,48 @@ function component_filecopy_valid = {
 
 # source and config are mutually exclusive, one is required.
 type structure_filecopy = {
-    'config' ? string      # Contents embedded in configuration
-    'source' ? string      # source file
+    @{The file content specified as a string.}
+    'config' ? string
+    @{The name of a source file already present on the machine to
+    use as the content for the managed file.}
+    'source' ? string
+    @{A command to execute if the file is modified. It is typically used to restart a service
+    but any valid command can be specified, including several commands separated by ';'.
+    If not specified, the file is updated but no command is executed.
+    As mentioned earlier, restart commands are executed after all files have been
+    updated and if several files specify the same restart command, it is executed once.}
     'restart' ? string
+    @{Permissions of the managed file. If not specified,
+     the default permissions on the system will be used.}
     'perms' ? string with match(SELF, '^[02-6]?[0-7]{3,3}$')
+    @{The userid of the file owner. It can also be a 'user:group' specification (like with chown).}
     'owner' ? string
+    @{The group of the file owner. It is ignored is owner is specified as 'user:group'.}
     'group' ? string
+    @{By default, the file content is converted to UTF8.
+    Define this property to 'true' to prevent this conversion.}
     'no_utf8' ? boolean
+    @{A boolean that defines if the restart command (if any defined).
+    must be executed even though the file was up-to-date (default behaviour is to execute the
+    restart command only if file content, permissions or owner/group has been changed).
+    Note: the global flag 'forceRestart' takes precedence if set to 'true'.}
```

Comment:
This is ambiguous. 

1. local = true, use global value (which could be true or false).
2. global value = true, use global value and ignore local value.

The second reading makes more sense but I haven't checked the code. If it's the second reading then I would say:

Note: this attribute is ignored if the global 'forceRestart' value is true.

Reply:
@ned21 the code states:
```
        # Check if the service must be restarted.
        # Default is to restart only if config file was changed.
        # Restart can be forced indepedently of changes, defining 'forceRestart'
        # properties globally or at the service level.
        my $service_restart = $changes;
        if ( ($file_config->{forceRestart} || $globalForceRestart ) ) {
            $service_restart = 1;
}
```
My perl is bad but I interpret this as your 2nd suggestion. I'll change it accordingly.